addons/escoria-core/game/esc_inputs_manager.gd

```diff
@@ -594,51 +606,89 @@ func _on_pause_menu_requested():
 	escoria.main.current_scene.game.pause_game()
 
 
-# Add the given item to the stack if not already in it.
-#
-# #### Parameters
-# - item: the item to add to the hover stack
-func hover_stack_add_item(item):
-	if item is ESCPlayer and not (item as ESCPlayer).selectable:
-		return
-	if not hover_stack.has(item):
-		hover_stack.push_back(item)
+class HoverStack:
+	
+	signal hover_stack_changed
+	
+	var hover_stack: Array = []
+	
+	# Add the given item to the stack if not already in it.
+	#
+	# #### Parameters
+	# - item: the item to add to the hover stack
+	func add_item(item):
+		if item is ESCPlayer and not (item as ESCPlayer).selectable:
+			return
+		if not hover_stack.has(item):
+			hover_stack.push_back(item)
+			sort()
+			emit_signal("hover_stack_changed")
+
+
+	# Add the items contained in given list to the stack if not already in it.
+	#
+	# #### Parameters
+	# - items: the items list (array) to add to the hover stack
+	func add_items(items: Array):
+		for item in items:
+			if escoria.action_manager.is_object_actionable(item.global_id):
+				add_item(item)
+
+
+	# Clean the hover stack
+	func clean():
+		for e in hover_stack:
+			if e == null or !is_instance_valid(e):
+				hover_stack.erase(e)
+				emit_signal("hover_stack_changed")
+	
+	
+	func pop_back():
+		var ret = hover_stack.pop_back()
+		if ret != null:
```

Comment:
Is it worth it to check using `is_instance_valid` instead?

Reply:
Fixed